test/opt/compact_ids_test.cpp

```diff
@@ -192,4 +192,84 @@ OpFunctionEnd
   EXPECT_THAT(disassembly, ::testing::Eq(expected));
 }
 
+// Test context consistency check after invalidating
+// CFG and others by compact IDs Pass.
+// Uses a GLSL shader with named labels for variety
+TEST(CompactIds, ConsistentCheck) {
+  const std::string input(R"(OpCapability Shader
+OpMemoryModel Logical GLSL450
+OpEntryPoint Fragment %main "main" %in_var_A %out_var_SV_TARGET
+OpExecutionMode %main OriginUpperLeft
+OpSource HLSL 600
+OpName %main "main"
+OpName %in_var_A "in.var.A"
+OpName %out_var_SV_TARGET "out.var.SV_TARGET"
+OpDecorate %in_var_A Location 0
+OpDecorate %out_var_SV_TARGET Location 0
+%void = OpTypeVoid
+%3 = OpTypeFunction %void
+%float = OpTypeFloat 32
+%v4float = OpTypeVector %float 4
+%_ptr_Input_v4float = OpTypePointer Input %v4float
+%_ptr_Output_v4float = OpTypePointer Output %v4float
+%in_var_A = OpVariable %_ptr_Input_v4float Input
+%out_var_SV_TARGET = OpVariable %_ptr_Output_v4float Output
+%main = OpFunction %void None %3
+%5 = OpLabel
+%12 = OpLoad %v4float %in_var_A
+%23 = OpVectorShuffle %v4float %12 %12 0 0 0 1
+OpStore %out_var_SV_TARGET %23
+OpReturn
+OpFunctionEnd
```

Comment:
I realize that any spirv with uncompacted IDs would have worked, but I thought I'd use this because it was unlike that used for many of the existing tests.